job-dsls/jobs/tools_provisioning_delete.groovy

```diff
@@ -0,0 +1,192 @@
+        wrappers {
+
+            // Adds timestamps to the console log.
+            timestamps()
+
+            // Adds timeout
+            timeout{
+                absolute(timeOutVar)
+            }
+
+            // Renders ANSI escape sequences, including color, to console output.
+            colorizeOutput()
+
+            // Deletes files from the workspace before the build starts.
+            preBuildCleanup()
+
+            // Loads file provider
+            configFileProvider {
+                managedFiles {
+                    configFile {
+                        fileId('openstack-upshift-rc-file')
+                        targetLocation('\$WORKSPACE/openrc.sh')
+                        variable('OPENRC_FILE')
+                    }
+                }
+            }
+
+            // Injects passwords as environment variables into the job.
+            injectPasswords {
+            // Injects global passwords provided by Jenkins configuration.
+                injectGlobalPasswords(true)
+                // Masks passwords provided by build parameters.
+                maskPasswordParameters(true)
+            }
+
+
+        }
+
+        // Adds custom properties to the job.
+        properties {
+
+            // Allows to configure job ownership.
+            ownership {
+
+                // Sets the name of the primary owner of the job.
+                primaryOwnerId("mbiarnes")
+
+                // Adds additional users, who have ownership privileges.
+                coOwnerIds("anstephe", "mnovotny", "almorale")
+            }
+        }
+
+        // Adds publishrs o job
+        publishers {
+            wsCleanup()
+        }
+
+        // Adds step that executes a shell script
+        steps {
+            shell(getScripts(jobAbr))
+        }
+    }
+}
+
+// def of variables
+String getDescription(String Abr){
+    switch(Abr) {
+        case "VerdaccioServ":
+            return "Destroys verdaccio-service machine from PSI (Upshift) OpenStack.<br>\n" +
+                    "Do NOT run unless you know what you are doing!"
+        case "DockerReg":
+            return "Destroys local docker-registry machine from PSI (Upshift) OpenStack.<br>\n" +
+                    "Do NOT run unless you know what you are doing!"
+        case "SmeeClient":
+            return "Destroys smee-client machine from PSI (Upshift) OpenStack.<br>\n" +
+                    "Do NOT run unless you know what you are doing!"
+        default:
+            return "Destroys cekit-cacher machine from PSI (Upshift) OpenStack.<br> \n" +
+                    "Do NOT run unless you know what you are doing!"
+    }
+}
+
+String getScripts(String Abr) {
+    switch(Abr) {
+        case "VerdaccioServ":
+            return ". \$OPENRC_FILE\n" +
+                    "# OS_PASSWORD gets injected from global passwords\n" +
+                    "\n" +
+                    "export FULL_NAME=\"verdaccio-service-\$INSTANCE_NUMBER\"\n" +
```

Comment:
This `FULL_NAME` variable could probably be parametrized and then all the scripts are the same.